src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,778 @@
+						{
+							var droppedOnNode = treeView.NodeFromContainer(droppedOnItem);
+							var selectedRoots = treeViewList.GetRootsOfSelectedSubtrees();
+							foreach (var node in selectedRoots)
+							{
+								var nodeIndex = treeViewList.FlatIndex(node);
+								if (treeViewList.IsFlatIndexValid(nodeIndex))
+								{
+									treeViewList.RemoveNodeFromParent(node);
+									((TreeViewNodeVector)droppedOnNode.Children).Append(node);
+								}
+							}
+
+							args.Handled = true;
+							treeViewList.OnDropInternal(args);
+						}
+						else
+						{
+							var droppedOnNode = treeView.NodeFromContainer(droppedOnItem);
+
+							// Remove the item that was dragged
+							int removeIndex = droppedNode.Parent.Children.IndexOf(droppedNode);
+
+							if (droppedNode != droppedOnNode)
+							{
+								((TreeViewNodeVector)droppedOnNode.Parent.Children).RemoveAt(removeIndex);
+
+								// Append the dragged dropped item as a child of the node it was dropped onto
+								((TreeViewNodeVector)droppedOnNode.Children).Append(droppedNode);
+
+								// If not set to true then the Reorder code of listview will override what is being done here.
+								args.Handled = true;
+								treeViewList.OnDropInternal(args);
+							}
+							else
+							{
+								args.AcceptedOperation = DataPackageOperation.None;
+							}
+						}
+					}
+				}
+			}
+			base.OnDrop(args);
+		}
+
+		protected override void OnDragOver(Windows.UI.Xaml.DragEventArgs args)
+		{
+			var treeView = AncestorTreeView;
+			if (treeView != null && !args.Handled)
+			{
+				var treeViewList = treeView.ListControl;
+				TreeViewItem draggedOverItem = this;
+				TreeViewNode draggedOverNode = treeView.NodeFromContainer(draggedOverItem);
+				TreeViewNode draggedNode = treeViewList.DraggedTreeViewNode;
+
+				if (draggedNode != null && treeView.CanReorderItems)
+				{
+					if (treeViewList.IsMutiSelectWithSelectedItems)
+					{
+						if (treeViewList.IsSelected(draggedOverNode))
+						{
+							args.AcceptedOperation = DataPackageOperation.None;
+							treeViewList.SetDraggedOverItem(null);
+						}
+						else
+						{
+							args.AcceptedOperation = DataPackageOperation.Move;
+							treeViewList.SetDraggedOverItem(draggedOverItem);
+						}
+					}
+					else
+					{
+						TreeViewNode walkNode = draggedOverNode.Parent;
+						while (walkNode != null && walkNode != draggedNode)
+						{
+							walkNode = walkNode.Parent;
+						}
+
+						if (walkNode != draggedNode && draggedNode != draggedOverNode)
+						{
+							args.AcceptedOperation = DataPackageOperation.Move;
+							treeViewList.SetDraggedOverItem(draggedOverItem);
+						}
+
+						treeViewList.UpdateDropTargetDropEffect(false, false, null);
+					}
+				}
+			}
+
+			base.OnDragOver(args);
+		}
+
+		protected override void OnDragEnter(Windows.UI.Xaml.DragEventArgs args)
+		{
+			TreeViewItem draggedOverItem = this;
+
+			args.AcceptedOperation = DataPackageOperation.None;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'args' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)